### ISSUE TITLE
Auto generated API.swift failed when field name is duplicated with difference pattern

### ISSUE THREAD

**Issue by Mangese** [apollographql/apollo-ios#1077]
I found some of Apollo\'s auto generate class from .graphql file might fail when field name is same in difference pettern

Assume that I have graphql schema as below
```
article {
   created_time: Date
   createdTime: Date (deprecated)
}
```

on compile, it will show an error "Definition conflicts with previous value" on API.swift. because Apollo always transform the snake-case field into camel-case 

The current behaviour makes sense. but I have a question that field transform is really necessary. because it might cause an inconsistency code between frontend & backend

**Comment by designatednerd** [comment]
Oof. Yeah I'm not sure there's a great way to opt out of that transformation in the current codegen. Best workaround I can advise is to alias one of the field names so they don't match. I'll definitely be keeping this in mind as I work on Swift Codegen.

**Comment by calvincestari** [comment]
Hi 👋🏻 - this issue should be resolved in 1.0. 1.0 is currently in alpha releases with a full 1.0 release in H2 2022. Please take a look at the [latest alpha](https://github.com/apollographql/apollo-ios/releases) and let us know if this is still an issue for you.
